main.go

```diff
@@ -79,40 +85,62 @@ func (items Items) CreateJSON(filename string) error {
 		return fmt.Errorf("error writing file `%s`: %v\n", filename, err)
 	}
 
+	log.WithFields(log.Fields{
+		"function": "CreateJSON()",
+	}).Info("writing to JSON finished successfully.")
+
 	return nil
 }
 
 func main() {
 	var items Items
-	// Load env variables
-	defer func() {
-		if err := recover(); err != nil {
-			fmt.Printf("recieved an error: %s\n", err)
-		}
-	}()
 
+	// Load env variables
 	if err := godotenv.Load(".env.dev"); err != nil {
 		if !errors.Is(err, os.ErrNotExist) {
-			panic(err)
+			log.WithFields(log.Fields{
+				"function": "main()",
+			}).Error("init env variables error occurred: ", err)
+			return
 		}
-		fmt.Printf("error loading env variables\n")
+
+		log.WithFields(log.Fields{
+			"function": "main()",
+		}).Error("os.ErrNotExist occurred while loading env variables: ", err)
 		return
 	}
 
+	log.WithFields(log.Fields{
+		"function": "main()",
+	}).Info("env file opened successfully")
+
 	// Parse xml file
 	if err := items.ParseXML(os.Getenv("XML_FILE_NAME")); err != nil {
 		if !errors.Is(err, os.ErrNotExist) {
-			panic(err)
+			log.WithFields(log.Fields{
```

Comment:
instead of adding "function" attribute to each log line it's much easier to pre-define a logger object for this specific method, e.g: 
```
func myFunction() {
   logger := logrus.New().WithField("method", "myFunction")

   ...
   logger.Debug("my debug message")
}
```